app/assets/javascripts/angular/directives/graph_chart.js

```diff
@@ -11,6 +12,8 @@ angular.module("Prometheus.directives").directive('graphChart', ["$location", "W
       var graphData = [];
       var annotationData = [];
 
+      element.on("click", ".legend .line", storeLegendState);
```

Comment:
Previously, storing legend state was tied to manually refreshing the graph, as that was where graph legend state was lost. However, changing the legend state only triggers an update within rickshaw, not within our code, so the it was possible for a user to change the legend state, request a shortlink, and upon visiting the shortlink find that the legend state wasn't persisted.

now, any time the legend is changed, that state is stored.